mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefullArScreen.kt

```diff
@@ -1,17 +1,42 @@
 package ch.epfl.sdp.mobile.state
 
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.*
 import androidx.compose.ui.Modifier
-import ch.epfl.sdp.mobile.application.chess.engine.Game
-import ch.epfl.sdp.mobile.ui.ar.ArScreen
+import ch.epfl.sdp.mobile.application.authentication.AuthenticatedUser
+import ch.epfl.sdp.mobile.ui.game.ar.ArChessBoardScreen
 
+/**
+ * A composable that make [ArChessBoardScreen] stateful
+ *
+ * @param id the identifier for the match.
+ * @param modifier the [Modifier] for the composable.
+ */
 @Composable
 fun StatefulArScreen(
+    user: AuthenticatedUser,
+    id: String,
     modifier: Modifier = Modifier,
 ) {
 
-  // FIX ME : Take the game for a give id
-  val game = Game.create()
+  val facade = LocalChessFacade.current
+  val scope = rememberCoroutineScope()
+  val match = remember(facade, id) { facade.match(id) }
 
-  ArScreen(game, modifier)
+  // FIXME : Not the cleanest code, but the faster way to deal with state with different UI. The AR
+  //  Screen didn't need all methods in [SnapshotChessBoardState]. The cleaner way is to have a
+  //  simpler snapshot that contains only information that will be needed by the
+  //  [ArChessBoardScreen]
```

Comment:
Maybe adding an issue for this could be a good idea.

Reply:
Done #262